modules/core/src/main/scala/org/tessellation/config/types.scala

```diff
@@ -41,8 +53,72 @@ object types {
   case class SnapshotConfig(
     heightInterval: NonNegLong,
     globalSnapshotPath: Path,
-    fallbackTriggerTimeout: FiniteDuration,
+    timeTriggerInterval: FiniteDuration,
     inMemoryCapacity: NonNegLong
   )
 
+  case class SoftStakingConfig(
+    address: Address,
+    nodes: NonNegInt,
+    startingOrdinal: SnapshotOrdinal,
+    softNodesPercentage: Percentage,
+    testnetNodes: NonNegInt,
+    testnetAddress: Address
+  )
+
+  case class DTMConfig(
+    address: Address,
+    monthly: Amount
+  )
+
+  case class StardustConfig(
+    address: Address,
+    percentage: Percentage
+  )
+
+  case class RewardsConfig(
+    epochs: PosInt,
+    epochDurationInYears: PosDouble,
+    totalPool: Amount,
+    baseEpochReward: Amount,
+    softStaking: SoftStakingConfig,
+    dtm: DTMConfig,
+    stardust: StardustConfig,
+    rewardsPerEpoch: NonEmptyMap[EpochProgress, Amount]
+  )
+
+  object RewardsConfig {
+
+    val default = RewardsConfig(
+      epochs = 4,
+      epochDurationInYears = 2.5,
+      totalPool = Amount(160000000000000000L),
+      baseEpochReward = Amount(85333333320000000L),
+      softStaking = SoftStakingConfig(
+        address = Address("DAG77VVVRvdZiYxZ2hCtkHz68h85ApT5b2xzdTkn"),
+        nodes = 0,
+        startingOrdinal = SnapshotOrdinal(0L),
+        softNodesPercentage = 40,
+        testnetNodes = 217,
+        testnetAddress = Address("DAG0qE5tkz6cMUD5M2dkqgfV4TQCzUUdAP5MFM9P")
+      ),
+      dtm = DTMConfig(
+        address = Address("DAG0Njmo6JZ3FhkLsipJSppepUHPuTXcSifARfvK"),
+        monthly = Amount(200000000000000L)
+      ),
+      stardust = StardustConfig(
+        address = Address("DAGSTARDUSTCOLLECTIVEHZOIPHXZUBFGNXWJETZVSPAPAHMLXS"),
+        percentage = 10
+      ),
+      rewardsPerEpoch = NonEmptyMap(
```

Comment:
`NonEmptyMap` isn't needed here. `SortedMap` is sufficient.

Reply:
I assumed that there must be at least 1 epoch. I can make it `SortedMap` and just mint no rewards if that's the case.
Having `NonEmptyMap` was just fitting better for expressing the business case.

Reply:
I don't see any reason why the type should guarantee that there is at least one epoch.